events/models/events.py

```diff
@@ -173,6 +173,29 @@ def update_event_searchable(event):
 
     searchable.save()
 
+def delete_event_searchable(event):
+    site = Site.objects.get(id=1)
+    event_url = "https://%s%s" % (site.domain, event.get_absolute_url())
+    origin_url = "https://%s%s" % (site.domain, reverse('searchables'))
+
+    md5 = hashlib.md5()
+    federation_url = event_url.split('/')
+    federation_node = '/'.join(federation_url[:3])
+    federation_id = '/'.join(federation_url[:5])
+    md5.update(bytes(federation_id, 'utf8'))
+    event_uri = federation_node + '/' + md5.hexdigest()
+
+    try:
+        searchable = Searchable.objects.get(event_uri=event_uri)
+    except:
+        searchable = Searchable(event_uri)
```

Comment:
You don't need to create the Searchable if you don't find one, so all of this is unnecessary. Just move the searchable.delete() up into the try block, and delete everything in the except block (just put 'pass' in there to avoid errors)

Reply:
Okay, will do. THanks